### ISSUE TITLE
[Fleet]: Unable to Uninstall Agent After Upgrade from 8.13.4 to 8.15.0 BC2

### ISSUE THREAD

**Issue by harshitgupta-qasource** [elastic/elastic-agent#5137]
**Kibana Build details:**
```
VERSION: 8.15.0 BC2
BUILD: 76111
COMMIT: 8d6510f30e7a52246ccebbf7d913a59b498c3bc2
```

**Host OS and Browser version:** All, All

**Preconditions:**
1. 8.15.0 BC2 Kibana Cloud environment should be available.
2. 8.13.4 agent should be deployed
3. Elastic Defend should be added to policy.

**Steps to reproduce:**

1. Navigate to Agents Tab

2. Upgrade the agent from 8.13.4 to 8.15.0.

3. Wait for the agent to upgrade and then uninstall Agent After Upgrade from 8.13.4 to 8.15.0 BC2.

4. Observe agent upgrade gets failed.

**Expected:**
- User should be able to Uninstall Agent After Upgrade from 8.13.4 to 8.15.0.

**Screenshot:**
- Windows
![image (7)](https://github.com/user-attachments/assets/19613bf0-ff19-40d4-a5ef-30653a2d5f11)

- Ubuntu
![image](https://github.com/user-attachments/assets/6eab1729-abe9-4403-8929-cc255a1ba4ef)

**Agents Logs:**
- Windows
[elastic-agent-diagnostics-2024-07-16T11-26-29Z-00(1).zip](https://github.com/user-attachments/files/16247905/elastic-agent-diagnostics-2024-07-16T11-26-29Z-00.1.zip)

- Linux
[elastic-agent-diagnostics-2024-07-16T11-26-29Z-00.zip](https://github.com/user-attachments/files/16247899/elastic-agent-diagnostics-2024-07-16T11-26-29Z-00.zip)

**Comment by harshitgupta-qasource** [comment]
@amolnater-qasource Kindly review

**Comment by amolnater-qasource** [comment]
Secondary Review for this ticket is Done.

**Comment by jlind23** [comment]
Hey @harshitgupta-qasource what was the system integration version you were using?

**Comment by jlind23** [comment]
This looks related to a change that Pierre recently introduced in https://github.com/elastic/integrations/pull/10471 so if you are able to downgrade the system integration to the previous version `1.59.0` this would allow us to rule this out.

**Comment by kpollich** [comment]
@jsoriano - Does anything about the implementation of this condition for the processors in the above look off to you? 

https://github.com/elastic/integrations/blob/b9a416c667a13e732c1a1f15eb722476c38b4ea5/packages/system/data_stream/auth/agent/stream/log.yml.hbs#L27-L39

Looking at other examples, I think this usage should be okay, but the errors above seem to indicate some kind of parsing error.

**Comment by pierrehilbert** [comment]
We already got this issue in the past, I'm investigating: https://github.com/elastic/go-ucfg/issues/196

**Comment by pierrehilbert** [comment]
I think the error is coming from here:
https://github.com/elastic/elastic-agent/blob/27dacc5bdaae99aabdfcefc74ef4ccd10627d3b8/internal/pkg/config/config.go#L179C4-L185C25

We are not using `ucfgOpts` and therefore we are not using `IgnoreCommas`

cc @fearful-symmetry as you spent time on the initial fix.

**Comment by fearful-symmetry** [comment]
yup, looks like we're missing the `IgnoreCommas` option from a few of the `Unpack` statements. Changing the statements to `err = subCfg.Unpack(&subArr, ucfg.ResolveNOOP, ucfg.IgnoreCommas)` should fix it.

**Comment by pierrehilbert** [comment]
@harshitgupta-qasource could you please test with the new System Integration package and confirm this is fixed?

**Comment by harshitgupta-qasource** [comment]
Hi @pierrehilbert 

Thanks for the looking into this issue.

We have re-validated this issue on the latest 8.15.0 BC4 Kibana cloud environment and found it **fixed now**.

Observations:
- User is able to Uninstall Agent After Upgrade from 8.13.4 to 8.15.0.
-  Available system  version integration is 1.60.0

**Build details:**
VERSION: 8.15.0
BUILD: 76261
COMMIT: 9d62937675e62265342e86d8f0db601dc75498b8


**Screen-shot**
![image](https://github.com/user-attachments/assets/d9d3e2a2-f762-4a9d-9512-b4762be22806)


Hence, we are closing this issue and marking as **QA: Validated**.

Thanks
